src/pint/toa.py

```diff
@@ -1917,3 +1920,73 @@ def read_toa_file(self, filename, process_includes=True, top=True):
                         newtoa.flags["to"] = self.cdict["TIME"]
                     self.toas.append(newtoa)
                     ntoas += 1
+
+
+def merge_TOAs(TOAs_list):
+    """Merge a list of TOAs instances and return a new combined TOAs instance
+
+    In order for a merge to work, each TOAs instance needs to have
+    been created using the same Solar System Ephemeris (EPHEM),  
+    the same reference timescale (i.e. CLOCK), and the same value of
+    .planets (i.e. whether planetary PosVel columns are in the tables
+    or not).
+
+    Parameters
+    ----------
+    TOAs_list : list of TOAs instances
+
+    Returns
+    -------
+    A new TOAs instance with all the combined and grouped TOAs
+    """
+    # Check each TOA object for consistency
+    ephems = [tt.ephem for tt in TOAs_list]
+    if len(set(ephems)) > 1:
+        log.warning(f"EPHEMs are inconsistent in merge_TOAs(): {ephems}\nNot merging.")
+        return None
```

Comment:
I think this is a clear case for an exception - the function is unable to do the specific thing you called it to do.

Reply:
Fair enough.  TypeError?